### PR TITLE
Remove ambiguity for kernel error

### PR DESCRIPTION
**Pull request by azarovh**
Also add test that ensures a specific reorg case works

### CODE REVIEW

File: chainstate/src/detail/ban_score.rs [hunk 1]
```diff
@@ -331,6 +331,8 @@ impl BanScore for ConsensusPoSError {
             ConsensusPoSError::StakeKernelHashTooHigh => 100,
             ConsensusPoSError::TimestampViolation(_, _) => 100,
             ConsensusPoSError::NoKernel => 100,
+            ConsensusPoSError::KernelOutpointMustBeUtxo => 100,
+            ConsensusPoSError::MissingKernelUtxo => 0,
```
Comment (TheQuantumPhysicist) [inline chainstate/src/detail/ban_score.rs:335]: This should be 100, right?

File: chainstate/test-suite/src/tests/pos_accounting_reorg.rs [hunk 2]
```diff
@@ -848,3 +848,121 @@ fn in_memory_reorg_disconnect_spend_delegation_from_decommissioned(#[case] seed:
...
+        .with_stake_spending_key(staker_sk_2)
+        .with_vrf_key(vrf_sk_2)
+        .build();
+    let block_e_id = block_e.get_id();
+    tf.process_block(block_e.clone(), BlockSource::Local).unwrap();
+
+    assert_eq!(block_d_id, block_e.prev_block_id());
+    assert_eq!(block_e_id, tf.best_block_id());
+
+    // Submit block_c that was saved and check that it's valid and reorg happend because it's denser chain
```
Comment (TheQuantumPhysicist) [inline chainstate/test-suite/src/tests/pos_accounting_reorg.rs:961]: happend

File: chainstate/test-suite/src/tests/pos_accounting_reorg.rs [hunk 3]
```diff
@@ -848,3 +848,121 @@ fn in_memory_reorg_disconnect_spend_delegation_from_decommissioned(#[case] seed:
...
+    assert_eq!(block_a_id, *block_b_index.prev_block_id());
+
+    // Build block_c but do not process it
+    let block_c = tf
+        .make_pos_block_builder(&mut rng)
+        .with_block_signing_key(staker_sk.clone())
+        .with_stake_spending_key(staker_sk.clone())
+        .with_vrf_key(vrf_sk.clone())
+        .build();
+    let block_c_id = block_c.get_id();
```
Comment (TheQuantumPhysicist) [inline chainstate/test-suite/src/tests/pos_accounting_reorg.rs:927]: This should come from staker 2

File: chainstate/src/detail/ban_score.rs (hunk 1 above)
Comment (azarovh) [reply chainstate/src/detail/ban_score.rs]: you are right

**Pull request merged**